src/header-navigation/examples.js

```diff
@@ -0,0 +1,258 @@
+/*
+Copyright (c) 2018 Uber Technologies, Inc.
+
+This source code is licensed under the MIT license found in the
+LICENSE file in the root directory of this source tree.
+*/
+// @flow
+/* eslint-disable react/display-name*/
+
+import * as React from 'react';
+import {Button} from '../button';
+import {StatefulMenu as Menu, KEY_STRINGS} from '../menu';
+import {StatefulSelect as Search} from '../select';
+import {styled} from '../styles';
+import COLORS from '../select/examples-colors';
+import {HamburgerIcon} from './examples-icons';
+
+import {HeaderNavigation, ALIGN, NavigationItem, NavigationList} from './index';
+import tests from './examples-list';
+import {TYPE} from '../select';
+
+const Link = styled('a', props => ({
+  ':hover': {
+    color: props.$theme.colors.primary400,
+  },
+}));
+
+let ITEMS = [];
+for (let i = 0; i < 100; i++) {
+  ITEMS.push({label: `Item ${i}`});
+}
+class MenuContainer extends React.Component<
+  {isOpen: boolean, align: string},
+  {isOpen: boolean},
+> {
+  static defaultProps = {
+    isOpen: false,
+    align: '',
+  };
+  state = {
+    isOpen: this.props.isOpen,
+  };
+  toggleMenu() {
+    this.setState({
+      isOpen: !this.state.isOpen,
+    });
+  }
+  render() {
+    return (
+      <React.Fragment>
+        <div
+          tabIndex="0"
+          aria-label="Navigation menu"
+          style={{lineHeight: 'initial'}}
+          onKeyDown={e => {
+            switch (e.key) {
+              case KEY_STRINGS.ArrowDown:
+              case KEY_STRINGS.Space:
+              case KEY_STRINGS.Enter:
+                this.toggleMenu();
+                return;
+            }
+          }}
+          onClick={this.toggleMenu}
+        >
+          {HamburgerIcon()}
+        </div>
+        {this.state.isOpen && (
+          <Menu
+            items={ITEMS}
+            overrides={{
+              List: {
+                style: {
+                  cursor: 'pointer',
+                  position: 'absolute',
+                  height: '90vh',
+                  right: this.props.align === ALIGN.right ? '0' : 'initial',
+                },
+              },
+            }}
+          />
+        )}
+      </React.Fragment>
+    );
+  }
+}
+
+export default {
+  [tests.HEADER_NAVIGATION_WITH_MENU_ELEMENT]: () => {
+    return (
+      <React.Fragment>
+        <div style={{marginTop: '50px'}} />
+        <HeaderNavigation>
+          <NavigationList $align={ALIGN.left}>
+            <NavigationItem>
+              <MenuContainer />
+            </NavigationItem>
+            <NavigationItem>Uber</NavigationItem>
+          </NavigationList>
+          <NavigationList $align={ALIGN.center} />
```

Comment:
So in my flexbox styles if all element are `left` or `right` docked flexbox still docks them together. I had to add this `center` to avoid it. I checked flexbox implementation and didn't find other solution. So there has to be at least one element, that is able to `flexGrow`.
![image](https://user-images.githubusercontent.com/1851469/48811531-83c7fb80-ed2e-11e8-8ae6-0ccc5bce8eb0.png)

![image](https://user-images.githubusercontent.com/1851469/48811553-9cd0ac80-ed2e-11e8-8f18-09655e067417.png)

![image](https://user-images.githubusercontent.com/1851469/48811605-def9ee00-ed2e-11e8-8b70-e37307a1bb37.png)